### PR TITLE
Update EventType.java - using varargs

### PR DESCRIPTION
**Pull request by siddhpant**
Issue: Consider using varargs for methods or constructors which take an array the last parameter. (x52)

One permalink: http://sonar.xwiki.org/issues/search#issues=a104439b-a2f0-4abe-87a5-bd132d3b2564

### CODE REVIEW

**Comment by tmortagne** [comment]
Ideally you should also update the code that use those methods to pass vararg instead of creating an array.

**Comment by siddhpant** [comment]
vararg is backward compatible, I had a chat on the IRC.

04/01/2018 6:55 pm को "Thomas Mortagne" <notifications@github.com> ने लिखा:

**Comment by tmortagne** [comment]
I know it's backward compatible, I did not suggested that you broken any API. I'm just saying that there is code in the same repository which could use those new APIs better now and would be nice to update it.

**Comment by siddhpant** [comment]
Okay, I found two files which import EventType.

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/QueueListener.java [hunk 1]
```diff
@@ -54,13 +54,13 @@
         /**
          * The parameters of the event.
          */
-        public Object[] eventParameters;
+        public Object... eventParameters;
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/QueueListener.java]: This is not valid Java.

**Comment by tmortagne** [comment]
Looks like all use go trough Event constructor so fixing it should be enough.

**Comment by siddhpant** [comment]
My bad. I was in a middle of something and also I was on mobile. Next time I would take care.

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/QueueListener.java (hunk 1 above)
Comment (siddhpant) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/QueueListener.java]: Man that's too obvious.

**Comment by siddhpant** [comment]
Done.

**Comment by tmortagne** [comment]
Thanks for your contribution !

**Pull request merged**